src/coreComponents/physicsSolvers/wavePropagation/AcousticWaveEquationSEMKernel.hpp

```diff
@@ -87,6 +87,47 @@ struct PrecomputeSourceAndReceiverKernel
     return false;
   }
 
+
+  GEOSX_HOST_DEVICE
+  static real64
+  evaluateRicker( real64 const & time_n,
+                  real64 const & f0,
+                  localIndex const & order )
+  {
+    real64 const o_tpeak = 1.0/f0;
+    real64 pulse = 0.0;
+    if((time_n <= -0.9*o_tpeak) || (time_n >= 2.9*o_tpeak))
+    {
+      return pulse;
+    }
+
+    constexpr real64 pi = M_PI;
+    real64 const lam = (f0*pi)*(f0*pi);
+
+    switch( order )
+    {
+      case 2:
+      {
+        pulse = 2.0*lam*(2.0*lam*(time_n-o_tpeak)*(time_n-o_tpeak)-1.0)*exp( -lam*(time_n-o_tpeak)*(time_n-o_tpeak));
+      }
+      break;
+      case 1:
+      {
+        pulse = -2.0*lam*(time_n-o_tpeak)*exp( -lam*(time_n-o_tpeak)*(time_n-o_tpeak));
+      }
+      break;
+      case 0:
+      {
+        pulse = -(time_n-o_tpeak)*exp( -2*lam*(time_n-o_tpeak)*(time_n-o_tpeak) );
```

Comment:
@mndiaye24 where does the order 0 Ricker come from? Shouldn't it be just a Gaussian?

Reply:
Here order n stand for n-th derivative of the Gaussian. So for order = 0, that mean we are using  a Gaussian

Reply:
Thank you Mamadou! In this case I think the formula for "case 0" should be changed, removing the factor "2" in the exponential and removing the pre-factor (time_n-o_tpeak), do you agree?